os/kernel/pthread/pthread_mutexinit.c

```diff
@@ -156,6 +156,7 @@ int pthread_mutex_init(FAR pthread_mutex_t *mutex,
 		if (status != OK) {
 			ret = get_errno();
 		}
+		mutex->sem.flags |= FLAGS_SEM_MUTEX;
```

Comment:
1. Is it ok to set it regardless of the status value?
2. Regardless of this PR, this function looks strange.. 
    1. This line and sem_setprotocol are executed regardless of result of sem_init.
    2. The sem_setprotocol overwrites the status value (sem_init result is removed..)